module/commands/lezioni.py

```diff
@@ -2,21 +2,54 @@
 """/lezioni command"""
 import logging
 import re
+import requests
+from io import BytesIO
+from bs4 import BeautifulSoup
 from typing import Tuple, Optional
 from telegram import InlineKeyboardButton, InlineKeyboardMarkup, Update, CallbackQuery
 from telegram.ext import CallbackContext
 from module.data import Lesson
-from module.shared import check_log, send_message
+from module.shared import check_log, send_message, read_md
 from module.data.vars import TEXT_IDS, PLACE_HOLDER
 from module.utils.multi_lang_utils import get_locale, get_locale_code
 
+HREF_TOKEN = "sites/default"
+DMI_LINK = read_md("informatica_link")
+
 logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s', level=logging.INFO)
 logger = logging.getLogger(__name__)
 
 
+def get_url(courses: str) -> str:
+    course_index = courses.find("L-31")
+    if course_index != -1:
+        token_pos = courses.find(":", course_index)
+        endl_index = courses.find("\n", course_index)
+
+        if token_pos != -1:
+            main_link = courses[token_pos + 1:endl_index]
+    return main_link
+
+
+def get_orario_file() -> Optional[bytes]:
```

Comment:
Valuterei l'inserimento di una cache in questa funzione. Il sito potrebbe essere temporaneamente non disponibile ma difficilmente il file degli orari cambia da un giorno all'altro, per cui potremmo fare in modo che il file venga scaricato solo se non presente o se è passato tot tempo dall'ultimo download.

Alla richiesta dell'utente viene quindi inviato il file presente in cache. Questo dovrebbe non solo velocizzare la risposta ma evitare il pericolo di DDoS, cosa da considerare se ogni esecuzione di /lezione porta ad un download dal sito.

Reply:
@Picred se te la senti, la puoi implementare anche in questa PR la cache, se no anche in una PR separata ovviamente, le PR meglio tenerle sempre "piccole", ma è uguale alla fine in questo caso 👍

Reply:
Si condivido, il mio è un suggerimento ma se per adesso non c'è tempo va benissimo la versione attuale.

Reply:
Effettivamente è un'ottima idea. L'unica cosa è che mi servirebbe sapere "cosa" suggerite di usare come cache. Al momento non mi viene nulla in mente forse perchè non conosco benissimo il codice sorgente ma se è "immediato" potrei anche farlo qui stesso, altrimenti si potrebbe apre una issue a riguardo

Reply:
Basterebbe secondo me scaricare il file in locale e mandare quello, tenendo magari in storage quando è stato scaricato l'ultima volta. Se è presente e non è passato troppo tempo manda il file locale, in alternativa lo scarica e lo manda solo dopo.

Questo è l'approccio "lazy" e credo sia il più semplice da implementare, ha il drawback che una chiamata ogni tanto verrà rallentata e sarà quella in cui è necessario scaricare il file. L'alternativa è scrivere un cronjob che, per esempio, ogni giorno alla stessa ora scarica il file dal sito.

Nel caso in cui sia necessario scaricare un file nuovo ed il sito non sia raggiungibile, valuterei di mandare l'ultima versione disponibile in locale mandando però un messaggio d'alert all'utente.